bootstrap.sh

```
@@ -14,6 +14,7 @@ kubectl apply -f .infrastructure/app/clusterIp.yml
 kubectl apply -f .infrastructure/app/nodeport.yml
 kubectl apply -f .infrastructure/app/hpa.yml
 kubectl apply -f .infrastructure/app/deployment.yml
+kubectl apply -f .infrastructure/security/rbac.yml
```

Critical: The script applies the Deployment before the RBAC manifest. This can cause pods to start without the `secrets-reader` ServiceAccount/Role/RoleBinding present, preventing the pod from listing secrets as required by the task checklist. Move `kubectl apply -f .infrastructure/security/rbac.yml` so it runs before `kubectl apply -f .infrastructure/app/deployment.yml`. Also consider adding a rollout wait after applying the deployment, for example:

```sh
kubectl apply -f .infrastructure/security/rbac.yml
kubectl apply -f .infrastructure/app/deployment.yml
kubectl rollout status deployment/todoapp -n todoapp --timeout=120s
```

And add `set -e` at the top to fail fast on errors. This change is necessary for the validation steps (ensuring pods start with the correct ServiceAccount permissions).